cpp/src/arrow/ipc/ipc-read-write-test.cc

```diff
@@ -725,12 +772,24 @@ class TestTensorRoundTrip : public ::testing::Test, public IpcTestFixture {
     int32_t metadata_length;
     int64_t body_length;
 
+    const auto& type = checked_cast<const FixedWidthType&>(*tensor.type());
+    const int elem_size = type.bit_width() / 8;
+
     ASSERT_OK(mmap_->Seek(0));
 
     ASSERT_OK(WriteTensor(tensor, mmap_.get(), &metadata_length, &body_length));
 
+    const int64_t expected_body_length = elem_size * tensor.size();
+
+    // Body padded
+    ASSERT_EQ(expected_body_length, body_length);
```

Comment:
I'm probably misunderstanding this test, but wouldn't you have to call `mmap_.Tell` if you wanted to check the padding?

Reply:
The comment is incorrect. `WriteTensor` does not pad the tensor body at the moment -- this turned out to be a huge can of worms (I tried to add padding) because of the way the TensorFlow ops work. We might return and add padding later, but it was too much work for right now